---
title: "Asthma metagenomic analysis, prepare multiomics data"
author: "A. Sorokin"
date: '`r format(Sys.time(), "%d.%m.%Y")`'
output:
  pdf_document: 
    keep_tex: yes
    number_sections: yes
  html_document: default
params:
  format: !r if(opts_knit$get("rmarkdown.pandoc.to") == 'html') c('screen', 'print')
    else 'print'
header-includes:
- \usepackage[T2A]{fontenc}
- \usepackage[utf8]{inputenc}
- \usepackage[english,russian]{babel}
- \usepackage{grffile}
- \usepackage{rotating}
- \usepackage{caption}
- \usepackage{longtable}
- \usepackage{lscape}
---
```{r loadPackages, include=FALSE, cache=FALSE}
## load additional packages in this chunk
library(pander)
library(knitr)
library(data.table)
library(ggplot2)
library(xtable)
library(tidyr)
library(dplyr)
library(openxlsx)
library(DESeq2)
library(apeglm)
library(lubridate)
library(compositions)
options(lubridate.week.start=1)

```

```{r setup, include=FALSE, cache=FALSE}
## This chunk should contain global configuration commands.
## Use this to set knitr options and related things. Everything
## in this chunk will be included in an appendix to document the
## configuration used.
#output <- opts_knit$get("rmarkdown.pandoc.to")
knitr::opts_knit$set(stop_on_error = 2L)

## By default R code is only included in HTML versions of the report
## (where it can be collapsed). You can generate a PDF version
## using rmarkdown::pdf_document to get a copy for print. Extensive
## chunks of R code may or may not be desired in /hat setting. If you
## want them simply change the following arguments to `echo = TRUE`.
## In either case the default can be overwritten for individual chunks.
#opts_chunk$set(echo = output=="html")
#opts_chunk$set(warning = output=="html")
#opts_chunk$set(message = output=="html")

## Cache options
knitr::opts_chunk$set(cache=TRUE,warning=FALSE, echo = FALSE,message=FALSE)

## Figure options
## Set default figure format
#options(reportmd.figure.format=params$format)

## Set 'hide.fig.code' to FALSE to include code chunks that
## produce Figures in the output. Note that this affects all chunks
## that provide a figure caption.
knitr::opts_chunk$set(hold=TRUE, hide.fig.code=FALSE)

## Set up default plotting options for different formats.
## These can be overwritten for individual chunks
#interactiveFig()
#screenFig()
#printFig()

## Pander options
pander::panderOptions("digits", 3)
pander::panderOptions("table.split.table", 160)
#panderOptions("table.style", "grid")

## Configure Figure and Table lables
#options(figcap.prefix = "Figure", figcap.sep = ":", figcap.prefix.highlight = "**")
#options(tabcap.prefix = "Table", tabcap.sep = ":", tabcap.prefix.highlight = "**")

## Install required knitr hooks
#installHooks()
```

```{r functions, include=FALSE}
## Custom functions used in the analysis should go into this chunk.
## They will be listed in their own section of the appendix.

##==================== Functions ====================##
```

# Read data
```{r read.metadata}
cmdt<-fread('patients_metadata.csv')
```

```{r load.metagenomics.data}
brakenDt<-read.delim('bracken_braken300F_combined_reports.txt')
rownames(brakenDt)<-brakenDt$taxonomy_id
ccts<-t(compositions::clr(
  t(as.matrix(brakenDt[,grep('^KS.*_num',names(brakenDt))]))))
colnames(ccts)<-sub('^(KS[^_]+)_.+','\\1',colnames(ccts))

cts<-as.data.table(ccts)
cts$taxonomy_id<-rownames(ccts)
mcts<-melt(cts,id.vars='taxonomy_id')
mcts$view<-'metagenomics'
names(mcts)<-c('feature','sample','value','view')
```
```{r load.metabolomics.data}
matamdt<-read.csv('/metabolome_annotation.csv')
metab<-read.csv('metabolome_DataPreprocessed.csv')
rownames(metab)<-metab$Metabolite.ID
metab$Name<-sub('α','alpha',metab$Name)
metab$Name<-sub('Α','Alpha',metab$Name)
metab$Name<-sub('β','beta',metab$Name)
metab$Name<-sub('ß','beta',metab$Name)
metab$Name<-sub('Β','Beta',metab$Name)
metab$Name<-sub('γ','gamma',metab$Name)
metab$Name<-sub('Υ','Gamma',metab$Name)
metab$Name<-sub('δ','delta',metab$Name)
metab$Name<-sub('Δ','Delta',metab$Name)
metab$Name<-sub('′',"'",metab$Name)
metab$Name<-sub('±',"+-",metab$Name)
metab$Name<-sub('–',"-",metab$Name)
```
```{r calc.clr.metabolomics.data}
cmts<-t(compositions::clr(
t(as.matrix(metab[,grep('^(asthma|healthy).+',names(metab))]))))
mts<-as.data.table(cmts)
mts$Metabolite.ID<-rownames(cmts)
mmts<-melt(mts,id.vars='Metabolite.ID')
mmts$view<-'metabolomics'
idx<-match(mmts$variable,matamdt$Sample.Name)
mmts$variable<-matamdt$ID.participant[idx]
names(mmts)<-c('feature','sample','value','view')
```

```{r merge.datasets}
lmodt<-rbind(mcts,mmts)
fwrite(lmodt,'mofa2_datasetF.csv')
```

```{r make.clean.metabolomics}
imetab<-metab[metab$Name!='unidentified',]
rownames(imetab)<-imetab$Metabolite.ID
imts<-mts[mts$Metabolite.ID%in%imetab$Metabolite.ID,]
immts<-melt(imts,id.vars='Metabolite.ID')
immts$view<-'metabolomics'
idx<-match(immts$variable,matamdt$Sample.Name)
immts$variable<-matamdt$ID.participant[idx]
names(immts)<-c('id','sample','value','view')
idx<-match(immts$id,imetab$Metabolite.ID)
immts$feature<-imetab$Name[idx]
imcts<-mcts
names(imcts)[1]<-'id'
idx<-match(imcts$id,brakenDt$taxonomy_id)
imcts$feature<-brakenDt$name[idx]
lmodt<-rbind(imcts,immts)
fwrite(lmodt,'mofa2_datasetF_identified.csv')
```

```{r make.kegg.metabolomics}
kmetab<-metab[!is.na(metab$KEGG.ID),]
rownames(kmetab)<-kmetab$Metabolite.ID
kmts<-mts[mts$Metabolite.ID%in%kmetab$Metabolite.ID,]
kmmts<-melt(kmts,id.vars='Metabolite.ID')
kmmts$view<-'metabolomics'
idx<-match(kmmts$variable,matamdt$Sample.Name)
kmmts$variable<-matamdt$ID.participant[idx]
names(kmmts)<-c('id','sample','value','view')
idx<-match(kmmts$id,kmetab$Metabolite.ID)
kmmts$feature<-kmetab$Name[idx]
kmcts<-mcts
names(kmcts)[1]<-'id'
idx<-match(kmcts$id,brakenDt$taxonomy_id)
kmcts$feature<-brakenDt$name[idx]
lmodt<-rbind(kmcts,kmmts)
fwrite(lmodt,'mofa2_datasetF_kegg.csv')
```

# Appendix {.tabset}
## Functions
```{r functions, eval=FALSE, include=TRUE}
```

## Setup R
```{r setup, eval=FALSE}
```

## Versions
### Document version
```{r docVersion, echo=FALSE, results='asis', cache=FALSE}
cat(params$version)
```

### Session Info
```{r sessionInfo, echo=FALSE, results='asis', class='text', warning=FALSE}
si<-devtools::session_info()
cat('Platform\n\n')
pander::pander(si$platform)
cat('Packages\n\n')
knitr::kable(as.data.frame(si$packages)[,c('ondiskversion','loadedversion','date','source')],align = c('l','l'))
```

